google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java

```diff
@@ -2708,6 +2709,120 @@ Blob create(
    */
   List<Acl> listAcls(BlobId blob);
 
+  /**
+   * Creates a new HMAC Key for the provided service account, including the secret key. Note that
+   * the secret key is only returned upon creation via this method.
+   *
+   * <p>Example of creating a new HMAC Key.
+   *
+   * <pre>{@code
+   * ServiceAccount serviceAccount = ServiceAccount.of("my-service-account@google.com");
+   *
+   * HmacKey hmacKey = storage.createHmacKey(serviceAccount);
+   *
+   * String secretKey = hmacKey.getSecretKey();
+   * HmacKey.HmacKeyMetadata metadata = hmacKey.getMetadata();
+   * }</pre>
+   *
+   * @throws StorageException upon failure
+   */
+  HmacKey createHmacKey(ServiceAccount serviceAccount);
+
+  /**
+   * Lists HMAC keys for a given service account. Note this returns {@code HmacKeyMetadata} objects,
+   * which do not contain secret keys.
+   *
+   * <p>Example of listing HMAC keys, specifying max results and showDeletedKeys.
+   *
+   * <pre>{@code
+   * ServiceAccount serviceAccount = ServiceAccount.of("my-service-account@google.com");
+   *
+   * Page<HmacKey.HmacKeyMetadata> metadataPage = storage.listHmacKeys(serviceAccount, null, 10, true);
+   * for (HmacKey.HmacKeyMetadata hmacKeyMetadata : metadataPage.getValues()) {
+   *     //do something with the metadata
+   * }
+   * }</pre>
+   *
+   * @param serviceAccount the service account whose HMAC keys to list
+   * @param pageToken the page from which to start results
+   * @param maxResults the maximum amount of results that can be returned by this request
+   * @param showDeletedKeys whether to show keys with the DELETED state in the result
+   * @throws StorageException upon failure
+   */
+  Page<HmacKeyMetadata> listHmacKeys(
+      ServiceAccount serviceAccount, String pageToken, Long maxResults, boolean showDeletedKeys);
```

Comment:
Based on other RPCs the Storage library has parameters: `serviceAccountEmail`, `pageToken`, `showDeletedKeys`, and `maxResults` should be listed under an Options optional parameter because they are optional. The only required value is `project_id` which is provided by client credentials. 

Other HmacKey RPCs should have a similar Options parameter for the value userProject. At the moment it's ignored by the service but it's expected to be available. Given it would change the signature of the method I thought it'd be best to do it now rather than wait until later. (I'm open to opinions here).